logic.js

```diff
@@ -0,0 +1,27 @@
+class Logic{
+
+    checkMatch(card1, card2){
+        if($(card1).data().id === $(card2).data().id && $(card1).data().gridArea !== $(card2).data().gridArea){
+            return true
+        } else{
+            return false
+        }
+    }
+
+    checkEndGame(){
+        if($('img').length ===0){
```

Comment:
Shouldn't have access to the DOM, if you were trying to go for MVC. In general, I didn't notice the site complying to MVC throughout.

Again, can check `!$('img').length`